### PR TITLE
Revamp the wx spu2-x dialog.

### PR DESCRIPTION
**Pull request by arcum42**
Now that I've gone and spent a bunch of time on the GSdx dialog, the wx one I did for spu2-x felt like I could do it better. Here it what it currently looks like:
old-dialog

So, lets make this have a tabbed interface, combine all the extra dialogs into the main dialog, and add Ok/Cancel buttons.
spu2-x1
spu2-x2
spu2-x3

### CODE REVIEW

**Comment by tellowkrinkle** [comment]
Is it possible to make this a label and not a text box?
Also it's scrollable for some reason 
image

**Comment by arcum42** [comment]
@tellowkrinkle The control itself is on line 120 of wxConfig.cpp. If you fiddle with the size on that line, the scroll should go away if it's large enough (That's why the size was set. Otherwise, it was smaller and giving a scrollbar on my side, too.). There is a wxTE_NO_VSCROLL attribute you could stick in the list, but it said it was only implemented on MSW and Gtk.

I originally had it as three, very fiddly wxStaticText labels that liked to cut off the end of the text with "..." and such. I just got fed up with messing with them, and decided to use a control that was actually designed for larger amounts of text and set it as read-only.

**Comment by sonicfind** [comment]
Also, it's "more than." LOL

File: plugins/spu2-x/src/wx/wxConfig.cpp [hunk 1]
```diff
@@ -18,471 +18,486 @@
...
-        auto adv_dialog = new SoundtouchCfg::AdvDialog;
-        adv_dialog->Display();
-        wxDELETE(adv_dialog);
-    }
+	Update();
 }
 
-void Dialog::ResetToValues()
+Dialog::Dialog()
+	: wxDialog(nullptr, wxID_ANY, "SPU2-X Config", wxDefaultPosition, wxDefaultSize, wxCAPTION | wxCLOSE_BOX)
```
Comment (lightningterror) [inline plugins/spu2-x/src/wx/wxConfig.cpp:345]: If we are getting rid of the gsdx branding then we might as well update this one too.

**Comment by arcum42** [comment]
@sonicfind I suppose I could change the text. I'd just copied it directly from what both the earlier wx dialog and the gtk dialog before it had. Tightening it up a bit so it takes up less space might be a good idea...

@lightningterror Sure, I could do that.

**Comment by arcum42** [comment]
spu2-x4
@tellowkrinkle It's back to being a label. See how that looks.
@sonicfind Majorly revised the text.
@lightningterror Changed the name.

**Comment by RedDevilus** [comment]
Is that your font making SPU2 config be cut off a bit on top? That second line best aligned to center, seems a bit too aligned to the left. The buttons themselves are also close to other text. Otherwise looks good to me.

**Comment by arcum42** [comment]
@RedDevilus This is wxWidgets. If things look even remotely close to where you want them, it's pretty much a miracle, and takes endless fiddling with alignment tags and things. Positioning is fiddly enough on gtk, and then wx adds its own layer on top...

And, yeah, that's probably the current theme making the font slightly cut off on the title bar.

**Comment by RedDevilus** [comment]
Sorry to bother you with that, didn't think it was that fiddly. 🙇

**Comment by arcum42** [comment]
Seriously, sometimes I'm staring at half a dozen lines of code with "Centre()" on them, and wondering why the bloody widgets aren't centered. Earlier the labels were centered, but only going 3/4's of the way across the dialog. There's a reason why I'm totally along with the "lets get rid of wx" train, despite also working in it a lot on pcsx2.

It is possible to absolutely position widgets (like in onepad), but that's the sort of decision you make before making the dialog, and is liable to give you dpi issues across platforms.

Part of the trouble is the stable version of wxwidgets, 3.0, is ancient. The unstable version, 3.1, I think has all sorts of fixes, including things like dpi, but the linux distros don't package it because it's "unstable". Version 3.2 is likely to come out by the end of the year, so things may improve.

Or we could rewrite everything in a different gui, which would also improve things. This rewrite gives us something to base that one on, though.

File: plugins/spu2-x/src/wx/wxConfig.cpp [hunk 2]
```diff
@@ -18,471 +18,484 @@
...
-    m_sdl_box->Add(m_sdl_text, wxSizerFlags().Centre());
+	auto* adv_box = new wxStaticBoxSizer(wxVERTICAL, this, "Advanced");
 
-    for (int i = 0; i < SDL_GetNumAudioDrivers(); ++i)
-        m_sdl.Add(SDL_GetAudioDriver(i));
+	auto* babble_label = new wxStaticText(this, wxID_ANY, \
+	"For fine-tuning time stretching.\n"\
+	"Larger is better for slowdown, && smaller for speedup (60+ fps).\n"\
+	"All options in microseconds.",\
+	wxDefaultPosition, wxSize(200, 65), wxALIGN_CENTRE_HORIZONTAL);
```
Comment (tellowkrinkle) [inline plugins/spu2-x/src/wx/wxConfig.cpp:121]: It's cutting off the last line 
image

Could you try this?
```suggestion
	wxDefaultPosition, wxDefaultSize, wxALIGN_CENTRE_HORIZONTAL);
	babble_label->Wrap(300);
```

Comment (arcum42) [reply plugins/spu2-x/src/wx/wxConfig.cpp:121]: Here's what that winds up looking like on Linux.
spu2-x5
IIRC, this is where I was before I made those changes, with everything centered off to the left for some reason...

Comment (tellowkrinkle) [reply plugins/spu2-x/src/wx/wxConfig.cpp:121]: It stays centered if you remove the `.Expand()` when adding it to the sizer

Comment (arcum42) [reply plugins/spu2-x/src/wx/wxConfig.cpp]: Alright, got it. Committed.

**Comment by arcum42** [comment]
Seems like this is about ready to me. I'll go ahead and merge it.

**Pull request merged**